### PR TITLE
Issue 21220: improve error handling for botEnableCmdF function

### PR DESCRIPTION
**Pull request by ariyonaty**
#### Summary

Error handling for the `botEnableCmdF` function updated to return an error in case of failure. 
Update includes the use of `multierror.Error` for appending all errors and returning error if one occurs.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/21220

### CODE REVIEW

**Review by isacikgoz — APPROVED** [review]
Thanks for the contribution @ariyonaty LGTM!

File: commands/bot.go [hunk 1]
```diff
@@ -216,22 +217,27 @@ func botListCmdF(c client.Client, cmd *cobra.Command, args []string) error {
 
 func botEnableCmdF(c client.Client, cmd *cobra.Command, args []string) error {
 	users := getUsersFromUserArgs(c, args)
+
+	var result *multierror.Error
+
 	for i, user := range users {
 		if user == nil {
 			printer.PrintError(fmt.Sprintf("can't find user '%v'", args[i]))
+			result = multierror.Append(result, fmt.Errorf("can't find user '%v'", args[i]))
```
Comment (noxer) [inline commands/bot.go:226]: Please use `%q` to quote strings
```suggestion
			result = multierror.Append(result, fmt.Errorf("can't find user %q", args[i]))
```

File: commands/bot.go [hunk 2]
```diff
@@ -216,22 +217,27 @@ func botListCmdF(c client.Client, cmd *cobra.Command, args []string) error {
...
 			continue
 		}
 
 		bot, _, err := c.EnableBot(user.Id)
 		if err != nil {
 			printer.PrintError(fmt.Sprintf("could not enable bot '%v'", args[i]))
+			result = multierror.Append(result, fmt.Errorf("could not enable bot '%v'", args[i]))
```
Comment (noxer) [inline commands/bot.go:233]: Same here. Please also include the error with `%w`.
```suggestion
			result = multierror.Append(result, fmt.Errorf("could not enable bot %q: %w", args[i], err))
```

File: commands/bot_e2e_test.go [hunk 3]
```diff
@@ -198,7 +198,7 @@ func (s *MmctlE2ETestSuite) TestBotEnableCmd() {
 		s.Require().Nil(appErr)
 
 		err := botEnableCmdF(s.th.Client, &cobra.Command{}, []string{newBot.UserId})
-		s.Require().Nil(err)
+		s.Require().NotNil(err)
```
Comment (noxer) [inline commands/bot_e2e_test.go:201]: Please check this with `s.Require().Error(err)`, it makes the test messages much easier to read.

File: commands/bot_e2e_test.go [hunk 4]
```diff
@@ -209,7 +209,7 @@ func (s *MmctlE2ETestSuite) TestBotEnableCmd() {
 		printer.Clean()
 
 		err := botEnableCmdF(c, &cobra.Command{}, []string{"nonexistent-bot-userid"})
-		s.Require().Nil(err)
+		s.Require().NotNil(err)
```
Comment (noxer) [inline commands/bot_e2e_test.go:212]: Same as above

**Review by noxer — CHANGES_REQUESTED** [review]
Hi @ariyonaty, thank you very much for your contribution. I've got a few requests but otherwise it looks good.

**Comment by noxer** [comment]
Hey @ariyonaty, do you have questions about the proposed changes?

**Comment by ariyonaty** [comment]
@noxer - Appreciate the feedback.  
Addressed the comments in commit [5c1a6fb](https://github.com/mattermost/mmctl/pull/577/commits/5c1a6fbf5af85b37574df6ccdafc72750e1f37ec).

**Review by noxer — APPROVED** [review]
Looks good, thank you very much @ariyonaty. :rocket:

**Pull request merged**